spec/support/matchers/have_actions.rb

```diff
@@ -67,9 +67,7 @@ def unless_option_for(action)
 end
 
 def parsed_only_action(action)
-  only_option = if_option_for(action)[0]
-
-  "#{only_option.class}OptionParser".constantize.new(only_option).parse
+  if_option_for(action)[0]
```

Comment:
I'll be honest, I don't have confidence in my understanding of this file's metaprogramming, and I'd rather see us test the behaviors around the `before_actions` via controller specs than ensure a method is called.

Reply:
omg I didn't realize this file was parsing the source, and not just introspecting the controller in memory. but +1 would rather see something like `expect(controller).to receive(:some_action).and_call_original`

Reply:
It's passing around a bunch of stuff and has to call `instance_variable_get` to get private instance variables, which isn't ideal and it's hard to follow the process.  I don't mind it as kind of a simpler check if it was well-supported, but it's unfortunately not the case.

It is used pretty extensively though.